api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/ObservationValueTranslatorImplTest.java

```diff
@@ -0,0 +1,247 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+import java.util.Date;
+
+import org.hl7.fhir.r4.model.BooleanType;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.IntegerType;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.Type;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Concept;
+import org.openmrs.ConceptDatatype;
+import org.openmrs.Obs;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir2.api.translators.ConceptTranslator;
+import org.powermock.core.classloader.annotations.PrepareOnlyThisForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareOnlyThisForTest({ Context.class })
+public class ObservationValueTranslatorImplTest {
+	
+	private static final String CONCEPT_VALUE_UUID = "12345-abcde-54321";
+	
+	private static final String OBS_STRING = "An ingenious observation";
+	
+	@Mock
+	private ConceptTranslator conceptTranslator;
+	
+	private Obs obs;
+	
+	private ObservationValueTranslatorImpl obsValueTranslator;
+	
+	@Before
+	public void setup() {
+		obsValueTranslator = new ObservationValueTranslatorImpl();
+		obsValueTranslator.setConceptTranslator(conceptTranslator);
+		
+		obs = new Obs();
+	}
+	
+	@Test
+	public void toFhirResource_shouldConvertObsWithCodedValueToCodeableConcept() {
+		Concept concept = new Concept();
+		concept.setUuid(CONCEPT_VALUE_UUID);
+		obs.setValueCoded(concept);
+		
+		CodeableConcept codeableConcept = new CodeableConcept();
+		codeableConcept.setId(CONCEPT_VALUE_UUID);
+		when(conceptTranslator.toFhirResource(concept)).thenReturn(codeableConcept);
+		
+		Type result = obsValueTranslator.toFhirResource(obs);
+		
+		assertThat(result, notNullValue());
+		assertThat(result, instanceOf(CodeableConcept.class));
+		assertThat(result.getId(), equalTo(CONCEPT_VALUE_UUID));
+	}
+	
+	@Test
+	public void toFhirResource_shouldConvertObsWithDatetimeToDateTimeType() {
+		Date expected = new Date();
+		obs.setValueDate(expected);
```

Comment:
I think `new Date()` can be volatile

Reply:
It can be, but in this case, this is harmless, because we expect that the date returned from the translator is the same as the date provided to the translator. In these circumstances (where we're testing equality against a dynamic value) this test might actually be able to detect subtle issues that arise around daylight-savings, leap-years, etc.